### ISSUE TITLE
Path Matching with Wildcard Host

### ISSUE THREAD

**Issue by yeddahbi** [jcmoraisjr/haproxy-ingress#751]
**Description of the problem**

Hello there,

Is it possible to do path matching while using wildcard host using haproxy-ingress v0.10.5?
We are having trouble working with both. 
We have 2 ingresses with the same wildcard host that are pointing to the same backend. However, we need to rewrite to a target path for one of the ingresses. See example below. Note that we have tried to the same configuration with a regular hostname, and it worked fine.

Not sure I fully understand what is happening, but from what I understand, the rewrite never happens because the path mapping to path1 is not working since the map file contains the wildcard hostname and the mapping is done using the “map_beg” function that does string matching.
https://github.com/jcmoraisjr/haproxy-ingress/blob/release-0.10/rootfs/etc/haproxy/template/haproxy.tmpl#L333

**Expected behavior**
Path matching/routing would work using a wildcard host.

**Steps to reproduce the problem**
The generated configuration for backend looks like this:

```
backend example-net-namespace_webapi- example-net-service_srv-port
    mode http
    balance roundrobin
    acl https-request ssl_fc
    # path01 = *.example.net/v2
    # path02 = *.example.net/status
    http-request set-var(txn.pathID) base,lower,map_beg(/etc/haproxy/maps/_back_example-net-namespace_webapi- example-net-service_srv-port_idpath.map)
    http-request set-header X-Original-Forwarded-For %[hdr(x-forwarded-for)] if { hdr(x-forwarded-for) -m found }
    http-request del-header x-forwarded-for
    option forwardfor
    http-request replace-path ^/v2/(.*)$       /to/another/path/v2\1     if { var(txn.pathID) path01 }
    http-response set-header Strict-Transport-Security "max-age=15768000"
    server srv001 10.10.11.111:8443 weight 1 ssl verify none check inter 2s
    server srv002 10.10.11.112:8443 weight 1 ssl verify none check inter 2s

```
Here is the content of the map file :

```
/ # cat /etc/haproxy/maps/_back_example-net-namespace_webapi- example-net-service_srv-port_idpath.map
# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
# #
# #   HAProxy Ingress Controller
# #   --------------------------
# #   This file is automatically updated, do not edit
# #
#
*.example.net/status path02
*.example.net/v2/ path01
```

**Environment information**

HAProxy Ingress version: `v0.10.5`

**Command-line options**
 - args:
            - >-
              --configmap=example-net-namepsace/example-net-haproxy-ingress-controller
            - '--ingress-class=my-ingress-class-ws
            - '--sort-backends'
            - >-
              --default-backend-service=example-net-namespace/example-net-ingress-default-bknd

Ingress objects:
```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    ingress.kubernetes.io/backend-protocol: HTTPS
    ingress.kubernetes.io/rewrite-target: /to/another/path/v2
    kubernetes.io/ingress.class: my-ingress-class-ws
  name: example-net-v2-ingress
  namespace: example-net-namespace
spec:
  rules:
    - host: '*.example.net'
      http:
        paths:
          - backend:
              serviceName: example-net-service
              servicePort: 443
            path: /v2
  tls:
    - hosts:
        - '*.example.net'
      secretName: example-net-certificate
```
```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    ingress.kubernetes.io/backend-protocol: HTTPS
    kubernetes.io/ingress.class: my-ingress-class-ws
  name: example-net-status-ingress
  namespace: example-net-namespace
spec:
  rules:
    - host: '*.example.net'
      http:
        paths:
          - backend:
              serviceName: example-net-service
              servicePort: 443
            path: /status
  tls:
    - hosts:
        - '*.example.net'
      secretName: example-net-certificate
 ```

**Comment by yeddahbi** [comment]
Thanks for the quick feedback. Unfortunately, we are stuck on an older version of k8s (<1.14) and cannot upgrade to v0.11 for the time being. 
We can however quickly test a fix on v0.10 when its available.

**Comment by jcmoraisjr** [comment]
v0.10.6 was just released and fixes this issue. Closing.
